eclipse-sw360/sw360
# feat(api): create new endpoint update vulnerabilities of a component

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)


- Run API  update vulnerabilities of a component.

Issue: #1941

### How To Test?
- Run command line: 
```
$  curl 'https://sw360.org/api/components/17653524/vulnerabilities' -i -X PATCH \
    -H 'Content-Type: application/json' \
    -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOlsic3czNjAtUkVTVC1BUEkiXSwidXNlcl9uYW1lIjoiYWRtaW5Ac3czNjAub3JnIiwic2NvcGUiOlsiYWxsIl0sImV4cCI6MTY4NDQwNDg3NiwiYXV0aG9yaXRpZXMiOlsiUkVBRCIsIldSSVRFIl0sImp0aSI6IlZvSlJ0UDFQYnRGT1ozN1hoak5STEFsc2hVdyIsImNsaWVudF9pZCI6InRydXN0ZWQtc3czNjAtY2xpZW50In0.f3VsF_0GIdr2ujZoMMYpgk2SphmJc0VQA2UZIRBBAIPKwDUdv4LD0f7bQ15pwPCC7KS3wwKsVjlIlCChuGLwayC8Uq4HWWwdGwoIhyGUa-MZvaG8S7Q16jtnDwANXjy8esxe2MOKkO6GANtVD0GxOQ5_BAriPkasSAm_3NB8YBr6fQk1TFzc3Dxvl32yDkFOsRQvez8mDQouxWAHQjiPkqaBqMA7x1rkf42N4HXWD6KjD4q3yULF9zrMjrquIsMz1eOKNgZCe_MBlKvUI9JRLDN4S4aEBeniaMoWAlTd8-_XpiC0jhggrV-MVxHU4OkiTDW8njBC8UBMpVv_OreDLA' \
    -H 'Accept: application/hal+json' \
    -d '{
  "releaseVulnerabilityRelationDTOs" : [ {
    "externalId" : "12345",
    "releaseName" : "Angular 2.3.0"
  } ],
  "comment" : "Change status",
  "verificationState" : "NOT_CHECKED"
}'
```
- Expected Output
```
{
  "_embedded" : {
    "sw360:vulnerabilityDTOes" : [ {
      "externalId" : "12345",
      "title" : "12345_Title",
      "priority" : "2 - major",
      "releaseVulnerabilityRelation" : {
        "releaseId" : "3765276512",
        "vulnerabilityId" : "1333333333",
        "verificationStateInfo" : [ {
          "checkedOn" : "2018-12-18",
          "checkedBy" : "admin@sw360.org",
          "comment" : "Change status not checked",
          "verificationState" : "NOT_CHECKED"
        }, {
          "checkedOn" : "2023-05-18",
          "checkedBy" : "admin@sw360.org",
          "comment" : "Change status",
          "verificationState" : "NOT_CHECKED"
        } ],
        "matchedBy" : "matchedBy",
        "usedNeedle" : "usedNeedle"
      },
      "projectRelevance" : "IRRELEVANT",
      "comment" : "Lorem Ipsum",
      "projectAction" : "Update to Fixed Version",
      "intReleaseId" : "3765276512",
      "intReleaseName" : "Angular 2.3.0",
      "_links" : {
        "self" : {
          "href" : "https://sw360.org/api/vulnerabilities/12345"
        }
      }
    } ]
  },
  "_links" : {
    "curies" : [ {
      "href" : "https://sw360.org/docs/{rel}.html",
      "name" : "sw360",
      "templated" : true
    } ]
  }
}
```
